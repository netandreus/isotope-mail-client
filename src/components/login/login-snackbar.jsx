import React, {Component} from 'react';
import {connect} from 'react-redux';
import Snackbar from '../snackbar/snackbar';
import {translate} from 'react-i18next';

export class LoginSnackbar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {error, t} = this.props;
    let message;
    if (error === null) {
      message = '';
    } else {
      message = t(`login.errors.${error}`);
    }
    return (
      <Snackbar show={error !== null} message={message}/>
    );
  }
}
const mapStateToProps = state => ({
  error: state.application.errors.authentication
});

export default connect(mapStateToProps)(translate()(LoginSnackbar));
