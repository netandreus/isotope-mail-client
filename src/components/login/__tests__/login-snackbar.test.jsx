import React from 'react';
import {shallow} from 'enzyme/build/index';
import {LoginSnackbar} from '../login-snackbar';

describe('LoginSnackbar component test suite', () => {
  describe('Snapshot render', () => {
    test('Renders hidden', () => {
      // Given
      const error = null;

      // When
      const loginSnackbar = shallow(<LoginSnackbar error={error}/>);

      // Then
      expect(loginSnackbar).toMatchSnapshot();
    });
    test('Renders with error', () => {
      // Given
      const t = jest.fn(messageKey => messageKey);
      const error = 'BLACKLISTED';

      // When
      const loginSnackbar = shallow(<LoginSnackbar error={error} t={t}/>);

      // Then
      expect(loginSnackbar).toMatchSnapshot();
      expect(t).toHaveBeenCalledTimes(1);
    });
  });
});
