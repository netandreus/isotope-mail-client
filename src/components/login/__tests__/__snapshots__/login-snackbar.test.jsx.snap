// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginSnackbar component test suite Snapshot render Renders hidden 1`] = `
<Snackbar
  alignStart={false}
  buttonAction={[Function]}
  buttonLabel=""
  message=""
  show={false}
/>
`;

exports[`LoginSnackbar component test suite Snapshot render Renders with error 1`] = `
<Snackbar
  alignStart={false}
  buttonAction={[Function]}
  buttonLabel=""
  message="login.errors.BLACKLISTED"
  show={true}
/>
`;
